### DISCUSSION TITLE
Why is the output of Catch22 multivariate of size nbr_instances*22 and not nbr_instances*nbr_series*22 ?

### DISCUSSION THREAD

**Discussion by Sinnaeve** [alan-turing-institute/sktime#1815] [Q&A]
I tried to use Catch22 to extract features.
I followed the example from main/examples/catch22.ipynb
Using the multivariate data from this example I have an input of 40 instannces each with 6 series of 100 data points. (40x6x100)
I would expect 22 features per series so an output of 40 instances each with 6 sets of 22 features (40x6x22), but I have an output of size 40x22.
It's writed: "The default procedure will concatenate each column prior to transformation" in the notebook, does that mean that for each instances, the algorithm take the 6 series (for 6 dimensions/columns) and flatten them in a single 600 data point time series ?
Does the features extracted means something then ? Shouldn't the catch22 algo be applied indivudally to each univariate series individually for a multivariate time series input ?

**Comment by fkiraly** [comment]
Thanks for raising this, @Sinnaeve!

This seems like a bug - would you be able to open an issue and also paste the simplest possible code that reproduces this, plus what you expect to happen instead?

Regarding series-to-primitive transformers applied to multivariate series, there is discussion in this issue here:
https://github.com/alan-turing-institute/sktime/issues/1719
It´s not clear how the correct output should be formatted (but it should not be what you get).

Would appreciate your opinion on the options.

**Reply by Sinnaeve** [reply]
Thanks for your answer, I have opened an issue. I will think about the format I would expect.

**Comment by fkiraly** [comment]
This is now resolved in the newest release, the return has your expected format, and future multivariate transformers will comply with that too:
https://github.com/alan-turing-institute/sktime/issues/1826
